go.mod

```diff
@@ -48,3 +48,5 @@ require (
 	k8s.io/component-base v0.25.4 // indirect
 	k8s.io/kube-openapi v0.0.0-20220803162953-67bda5d908f1 // indirect
 )
+
+replace golang.org/x/text => golang.org/x/text v0.4.0
```

Comment:
Why do we need the replace?

Reply:
I see, it's a transitive dependency that has that requirement. I guess the potential concern is that version combination being untested.

Reply:
There's also an analysis that Jan did in https://kubernetes.slack.com/archives/C8EJ01Z46/p1669292513960349

Reply:
To address the compatibility issues, we could change replace clause from v0.4.0 to v0.3.8. However, according to semantic versioning, which is followed by golang, upgrade of MINOR is backward compatible.

The analysis seems to be correct, but upgrading seems like a more trustworthy approach.

Reply:
We discussed this at today's csi standup. It looks like go.sum lists all the possible versions that could be used, but that doesn't necessarily mean that the version is being used. So cve scanners that rely on go.sum may produce a lot of false positives.

https://stackoverflow.com/questions/70128040/content-of-go-sum-and-modules-really-used-by-a-go-application
https://stackoverflow.com/questions/69825533/why-does-go-sum-include-so-many-older-packages

Given that this change removed the entries from go.sum, but didn't actually change anything in the vendor directory (which reflects what's actually being used) confirms this.

What scanner are you using? It looks like trivy no longer relies on go.sum when the go version in go.mod is >= 1.17: https://github.com/aquasecurity/trivy/issues/1973. But dependabot still does not: https://github.com/dependabot/dependabot-core/issues/4740

Reply:
I used a commercial security analysis tool to detect vulns. It is a good point that presence of a dependency in go.sum does not necessarily mean that it's actually used. However, the affected version (v0.3.7) is explicitly listed in `go.mod` of node-driver-registrar (indirect dependency but still a dependency) (it's not go.sum issue that was mentioned).

I agree that it looks like the affected part of the code is not used (even indirectly) by node-driver-registrar. However, afaik security scanners standards are not designed to go into analysis on a function level, they stop at reporting affected package version. Yet, they are still useful tool.

I believe that there is a choice to be made here by maintainers. Every time a new vulnerability is discovered either affected dependency should be updated regardless if the node-driver-registrar is using the affected, or do the vulnerability security audit every time and somehow maintain list of such vulnerabilities which are believed to be not affecting the repo and make all users aware that their scans might produce higher-than-usual number of false-positives.

Reply:
In the go.mod, it looks like only 0.4.0 is listed: https://github.com/kubernetes-csi/node-driver-registrar/pull/251/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R42

Anyway, I think we should go ahead and merge this because it is a burden to have users manage exceptions in their scanners, but we should also notify the scanning tools of false-positives so that they can be improved.

Reply:
Yeah, you are actually right, I was looking at 2.6.1 and 2.6.2 tags. `0.4.0` is not on master b/c of https://github.com/kubernetes-csi/node-driver-registrar/pull/243 which I've just noticed. 

The above makes this PR redundant, thank you for the discussion though! I think it's important to recognize how vulns should be addressed.